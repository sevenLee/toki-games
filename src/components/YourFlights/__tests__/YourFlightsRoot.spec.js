import React from 'react'
import {Provider} from 'react-redux'
import YourFlightsRootContainer, {YourFlightsRoot} from '../components/YourFlightsRoot'
import {shallow, mount} from 'enzyme'
import configureStore from 'redux-mock-store'


describe('<YourFlightsRoot />', () => {
    it('should render YourFlightsRoot', () => {
        const wrapper = shallow(<YourFlightsRoot/>)
        console.log(wrapper.debug())
    })
})

describe('<YourFlightsRootContainer />', () => {
    let wrapper, store

    const initalState = {
        cheapFlights: {
            list: [],
            isLoading: false,
            sorting: '',
            descending: false,
            activePage: 1,
            filter: 'arrival',
            filterWords: ''
        },
        businessFlights: {
            list: [],
            isLoading: false,
            sorting: '',
            descending: false,
            activePage: 1
        },
        form: {
            flightForm: {
                values: {
                    flightType: "cheap"
                },
                initial: {
                    flightType: "cheap"
                },
                registeredFields: {
                    flightType: {},
                    departure: {},
                    arrival: {},
                    departureTime: {},
                    arrivalTime: {}
                },
                fields: {
                    flightType: {},
                    departure: {},
                    arrival: {},
                    departureTime: {},
                    arrivalTime: {}
                }
            }
        }
    }

    const mockStore = configureStore();
    store = mockStore(initalState)
    wrapper = mount( <Provider store={store}><YourFlightsRootContainer /></Provider> )

    it('should set form submit success with redux-form/SET_SUBMIT_SUCCEEDED', () => {
        wrapper.find('input[name="departure"]').simulate('change', {target: {value: 'Departure'}})
        wrapper.find('input[name="arrival"]').simulate('change', {target: {value: 'Arrival'}})
        wrapper.find('button[type="submit"]').getDOMNode().click()
        const action = store.getActions()
        expect(store.getActions()).toMatchSnapshot();
    })
})