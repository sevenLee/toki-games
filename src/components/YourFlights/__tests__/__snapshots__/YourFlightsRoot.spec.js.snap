// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YourFlightsRootContainer /> should set form submit success with redux-form/SET_SUBMIT_SUCCEEDED 1`] = `
Array [
  Object {
    "meta": Object {
      "form": "flightForm",
    },
    "payload": Object {
      "name": "flightType",
      "type": "Field",
    },
    "type": "@@redux-form/REGISTER_FIELD",
  },
  Object {
    "meta": Object {
      "form": "flightForm",
    },
    "payload": Object {
      "name": "departure",
      "type": "Field",
    },
    "type": "@@redux-form/REGISTER_FIELD",
  },
  Object {
    "meta": Object {
      "form": "flightForm",
    },
    "payload": Object {
      "name": "arrival",
      "type": "Field",
    },
    "type": "@@redux-form/REGISTER_FIELD",
  },
  Object {
    "meta": Object {
      "form": "flightForm",
    },
    "payload": Object {
      "name": "departureTime",
      "type": "Field",
    },
    "type": "@@redux-form/REGISTER_FIELD",
  },
  Object {
    "meta": Object {
      "form": "flightForm",
    },
    "payload": Object {
      "name": "arrivalTime",
      "type": "Field",
    },
    "type": "@@redux-form/REGISTER_FIELD",
  },
  Object {
    "meta": Object {
      "form": "flightForm",
    },
    "payload": Object {
      "error": undefined,
      "syncErrors": Object {
        "arrival": "Value is required",
        "departure": "Value is required",
      },
    },
    "type": "@@redux-form/UPDATE_SYNC_ERRORS",
  },
  Object {
    "type": "FETCH_CHEAP_FLIGHTS_ACTION",
  },
  Object {
    "type": "FETCH_BUSINESS_FLIGHTS_ACTION",
  },
  Object {
    "meta": Object {
      "field": "departure",
      "form": "flightForm",
      "persistentSubmitErrors": false,
      "touch": false,
    },
    "payload": "Departure",
    "type": "@@redux-form/CHANGE",
  },
  Object {
    "meta": Object {
      "field": "arrival",
      "form": "flightForm",
      "persistentSubmitErrors": false,
      "touch": false,
    },
    "payload": "Arrival",
    "type": "@@redux-form/CHANGE",
  },
  Object {
    "meta": Object {
      "fields": Array [
        "flightType",
        "departure",
        "arrival",
        "departureTime",
        "arrivalTime",
      ],
      "form": "flightForm",
    },
    "type": "@@redux-form/TOUCH",
  },
  Object {
    "type": "ADD_CHEAP_FLIGHT_ACTION",
  },
  Object {
    "error": false,
    "meta": Object {
      "fields": Array [],
      "form": "flightForm",
    },
    "type": "@@redux-form/SET_SUBMIT_SUCCEEDED",
  },
]
`;
